plugin/trino-ranger/src/main/java/io/trino/plugin/ranger/RangerSystemAccessControl.java

```diff
@@ -0,0 +1,1967 @@
+        implements SystemAccessControl
+{
+    private static final Logger LOG = LoggerFactory.getLogger(RangerSystemAccessControl.class);
+
+    public static final String RANGER_TRINO_DEFAULT_HADOOP_CONF = "trino-ranger-site.xml";
+    public static final String RANGER_TRINO_SERVICETYPE = "trino";
+    public static final String RANGER_TRINO_APPID = "trino";
+    public static final String RANGER_TRINO_DEFAULT_SERVICE_NAME = "dev_trino";
+    public static final String RANGER_TRINO_DEFAULT_SECURITY_CONF = "ranger-trino-security.xml";
+    public static final String RANGER_TRINO_DEFAULT_AUDIT_CONF = "ranger-trino-audit.xml";
+    public static final String RANGER_TRINO_DEFAULT_POLICYMGR_SSL_CONF = "ranger-policymgr-ssl.xml";
+
+    private final RangerBasePlugin rangerPlugin;
+    private final boolean useUgi;
+    private final RangerTrinoEventListener eventListener = new RangerTrinoEventListener();
+
+    @Inject
+    public RangerSystemAccessControl(RangerConfig config)
+    {
+        super();
+
+        setDefaultConfig(config);
+
+        Configuration hadoopConf = new Configuration();
+
+        if (config.getHadoopConfigPath() != null) {
+            URL url = hadoopConf.getResource(config.getHadoopConfigPath());
+
+            if (url == null) {
+                LOG.warn("Hadoop config {} not found", config.getHadoopConfigPath());
+            }
+            else {
+                hadoopConf.addResource(url);
+            }
+        }
+        else {
+            URL url = hadoopConf.getResource(RANGER_TRINO_DEFAULT_HADOOP_CONF);
+
+            LOG.info("Trying to load Hadoop config from {} (can be null)", url);
+
+            if (url != null) {
+                hadoopConf.addResource(url);
+            }
+        }
+
+        UserGroupInformation.setConfiguration(hadoopConf);
+
+        if (config.getKeytab() != null && config.getPrincipal() != null) {
+            String keytab = config.getKeytab();
+            String principal = config.getPrincipal();
+
+            LOG.info("Performing kerberos login with principal {} and keytab {}", principal, keytab);
+
+            try {
+                UserGroupInformation.loginUserFromKeytab(principal, keytab);
+            }
+            catch (IOException ioe) {
+                LOG.error("Kerberos login failed", ioe); // ERROR
+
+                throw new RuntimeException(ioe);
+            }
+        }
+
+        useUgi = config.isUseUgi();
+
+        RangerPluginConfig pluginConfig = new RangerPluginConfig(RANGER_TRINO_SERVICETYPE, config.getServiceName(), RANGER_TRINO_APPID, null, null, null);
+
+        pluginConfig.addResourceIfReadable(config.getAuditConfigPath());
+        pluginConfig.addResourceIfReadable(config.getSecurityConfigPath());
+        pluginConfig.addResourceIfReadable(config.getPolicyMgrSslConfigPath());
+
+        rangerPlugin = new RangerBasePlugin(pluginConfig);
+
+        try {
+            rangerPlugin.init();
+        }
+        catch (Throwable t) {
```

Comment:
We should not swallow exceptions like that. If this method fails I guess we should fail the Trino setup too.

Reply:
You are right. Fixed.